### ISSUE TITLE
Problem by using <rosparam_handler/utilities.hpp> with map and ostream

### ISSUE THREAD

**Issue by destogl** [cbandera/rosparam_handler#55]
Hi,

first of all thank You for great and very useful package. I am using this package for a more than a year I didn't have any major issues until now. Nevertheless the problem is probably not in Your package but in a way I am using it. I managed to pin-point the error, but I don't understand it. Hopefully someone here can help....

Using the current released version, master or devel branch from this repository I am getting following error regarding output of maps:
```
error: no match for ‘operator<<’ (operand types are ‘std::basic_ostream<char>’ and ‘const std::map<std::__cxx11::basic_string<char>, std::__cxx11::basic_string<char> >’)
```
[Here](https://travis-ci.org/KITrobotics/force_torque_sensor/builds/393529731?utm_source=github_status&utm_medium=notification) you can find complete Travis output.

The strange this is that if I am using commit 2a71b33 ([here](https://travis-ci.org/KITrobotics/force_torque_sensor) you can find the build)when <rosparam_handler/utilities.hpp> is not used in `Parameters.h.templated`-file there is no error. As far as I compared the code it seams to be compatible with, but moving some overrides in `utilities.hpp`....

**Comment by artivis** [comment]
Hi, thanks for the great feedback !

Unfortunately I could not reproduce the issue with a small example (a couple parameters including a `std::map<std::string, std::string>`).  

However starting from your Travis link I went up to the https://github.com/KITrobotics/force_torque_sensor/pull/4 and branch `use-upstrea-rosparam-handler`. On this branch, the cfg file [`calibration.params`](https://github.com/KITrobotics/force_torque_sensor/blob/21e9b2172a027deb3de9bfd824d33373621b8ee1/cfg/Calibration.params#L8) has a [`force`](https://github.com/KITrobotics/force_torque_sensor/blob/21e9b2172a027deb3de9bfd824d33373621b8ee1/cfg/Calibration.params#L8) parameter defined as being a `std::map<std::string, std::string>` whereas it is a `std::map<std::string, double>` on your main branch.  

I do not know if this change was intended but I suspect the issue to come from there.

**Comment by artivis** [comment]
After changing `force` from `std::map<std::string, std::string>` to `std::map<std::string, double>`, the branch `use-upstrea-rosparam-handler` seems to compile (I still have a couple error due to extra compiler flags -> e.g. `force_torque_sensor_sim.cpp:24:1: error: no return statement in function returning non-void [-Werror=return-type]`).  
No more stream-related issues :+1:

**Comment by artivis** [comment]
Closing as this seems to be fixed.  
@destogl please feel free to re-open in case it is not.

**Comment by destogl** [comment]
Hi @artivis,

sorry not to answering earlier. I missed the notifications from your messages. 

> Unfortunately I could not reproduce the issue with a small example (a couple parameters including a std::map<std::string, std::string>).

This change was done because I was trying to find error myself and later forgot to revert it. Unfortunately the problem still exists...

Am I missing some compiler flags?

P.S. I am not able to re-open the issue...

**Comment by artivis** [comment]
Hi @destogl,
your issue could be related to #57 (or arise in a similar way). #57 is fixed with PR #58, maybe you could try the branch `fix/map_vec_streamop` and check if it solves your issue too ??
I'm re-opening this ticket in the mean time.

**Comment by destogl** [comment]
Hi @artivis,

thanks! This works!
